### ISSUE TITLE
Migrate to using Docsy as an npm module

### ISSUE THREAD

**Issue by SayakMukhopadhyay** [kubernetes/contributor-site#557]
We would like to move from using Docsy as a git submodule to an NPM dependency. This is the proposed approach for the docs website (kubernetes/website#48812) and we would like to do it here too, ideally after #531 is merged.

**Comment by jberkus** [comment]
Sounds good, let Contribex know when you want to get started.

**Comment by SayakMukhopadhyay** [comment]
I will work on it but I would like #531 to be merged first as I know that the PR works right now. Doing this before #531 would mean additional rework.

**Comment by sftim** [comment]
I think we should do this before #531; I just saw that the change doesn't actually preview OK @SayakMukhopadhyay. Adopting the approach planned for k/website should fix this issue and make upgrades easier.

**Comment by SayakMukhopadhyay** [comment]
@sftim I saw your [comment](https://github.com/kubernetes/website/pull/48726#issuecomment-2482896242) on the kubernetes/website#48726 PR. Are you considering using Hugo modules for managing the docsy themes?

**Comment by SayakMukhopadhyay** [comment]
> I think we should do this before #531; I just saw that the change doesn't actually preview OK @SayakMukhopadhyay. Adopting the approach planned for k/website should fix this issue and make upgrades easier.

The problem is that the current version of Docsy that is deployed didn't support themeing in either NPM or Go modules way. Git Submodules was the only option. So, this change cannot happen without migrating to at least 0.5.

**Comment by sftim** [comment]
Let's move to Docsy 0.5.1 (same approach as k/website). Then, continue onwards.

**Comment by SayakMukhopadhyay** [comment]
In fact, the change from 0.5 to 0.6 is trivial. Its 0.7 which brings the new bootstrap stuff. So, I am ready to split #531 into 2 PRs. I can rebase #531 to keep until 0.6 and a new PR to 0.11

**Comment by sftim** [comment]
> Let's move to Docsy 0.5.1 (same approach as k/website). Then, continue onwards.

Still my recommendation. 0.6 is fine; moving via 0.5.1 will make things overall easier for approvers.
